### PR TITLE
Support Mono-InternVL with PyTorch backend

### PR DESCRIPTION
**Pull request by wzk1015**
Support [Mono-InternVL](https://huggingface.co/OpenGVLab/Mono-InternVL-2B) with Pytorch backend by creating `internlm2_ve.py` and adding `is_mono` in `internvl.py`. The MoE structure is implemented with `vision_embedding_indexing` and `text_embedding_indexing`. Also fix some typos.

### CODE REVIEW

**Comment by grimoire** [comment]
OMG, we have so many typos...

**Comment by lvhan028** [comment]
Please add Mono-InternVL into test cases

**Comment by wzk1015** [comment]
One known bug is that Mono-InternVL encounters NaN when using float16 ([this issue](https://huggingface.co/OpenGVLab/Mono-InternVL-2B/discussions/9)), due to some numerical instability. We will fix this in the next version of model, but for the current version I think float16 is not supported. Should I add a note in `FP16/BF16` of `supported_models.md`?

I test KV INT 8 and INT 4 and they work fine. For W8A8 and W4A16 I suppose they are not supported, following InternVL2 PyTorchEngine.

**Comment by lvhan028** [comment]
> One known bug is that Mono-InternVL encounters NaN when using float16 ([this issue](https://huggingface.co/OpenGVLab/Mono-InternVL-2B/discussions/9)), due to some numerical instability. We will fix this in the next version of model, but for the current version I think float16 is not supported. Bfloat16 works fine. Should I add a note in `FP16/BF16` of `supported_models.md`?
> 
> I test KV INT 8 and INT 4 and they work fine. For W8A8 and W4A16 I suppose they are not supported, following InternVL2 PyTorchEngine.

Sure. A note about this situation is appreciated.

**Review by grimoire — APPROVED** [review]
LGTM

File: lmdeploy/pytorch/models/module_map.py [hunk 1]
```diff
@@ -149,6 +149,12 @@
     f'{LMDEPLOY_PYTORCH_MODEL_PATH}.internvl.InternVLChatModel'
 })
 
+# mono-internvl
+MODULE_MAP.update({
+    'InternLM2VEForCausalLM':
+    f'{LMDEPLOY_PYTORCH_MODEL_PATH}.internlm2_ve.InternLM2VEForCausalLM',
```
Comment (RunningLeon) [inline lmdeploy/pytorch/models/module_map.py:155]: may update this supported list as well
https://github.com/InternLM/lmdeploy/blob/2bed0182a8bac22e815ddfb6e00be1f680dea927/lmdeploy/pytorch/supported_models.py#L64

**Review by RunningLeon — APPROVED** [review]
LGTM

**Pull request merged**